src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -3648,26 +3649,24 @@ ntp_archiver::find_reupload_candidate(manifest_scanner_t scanner) {
                 "unexpected default re-upload candidate creation result");
           },
           [this, &run, &rw_fence, units = std::move(units)](
-            upload_candidate_with_locks& upload_candidate) mutable
+            segment_collector_stream& collector_stream) mutable
             -> find_reupload_candidate_result {
               if (
-                upload_candidate.candidate.content_length
-                  != run->meta.size_bytes
-                || upload_candidate.candidate.starting_offset
-                     != run->meta.base_offset
-                || upload_candidate.candidate.final_offset
-                     != run->meta.committed_offset) {
+                collector_stream.size != run->meta.size_bytes
+                || collector_stream.start_offset != run->meta.base_offset
+                || collector_stream.end_offset != run->meta.committed_offset) {
                   vlog(
                     _rtclog.error,
                     "Failed to make reupload candidate to match the run, "
-                    "candidate: "
-                    "{}, run: {}",
-                    upload_candidate.candidate,
+                    "candidate: {} run: {}",
+                    collector_stream,
                     run->meta);
-                  return {std::nullopt, std::nullopt, {}};
+                  return {};
               }
-
-              return {std::move(units), std::move(upload_candidate), rw_fence};
+              return {
+                .units = std::move(units),
+                .locks = std::move(collector_stream),
```

Comment:
maybe the name should be different? we're assigning 'stream' to 'locks' field which is kind of confusing